### PR TITLE
Added Mock of Minikube Delete Profiles Test portion.

### LINKED ISSUE #9795: Unit test TestDeleteAllProfiles fails: delete_test.go:203: ListProfiles length = 9, expected 8
TestDeleteAllProfiles fails but only after running:
"minikube start"

If "minikube delete" is run - test passes.  

**Steps to reproduce the issue:** 
1. make 
2.  out/minikube start
3. make test

**Full output of failed command:**
[opc@kubernetes1 minikube]$ make test
MINIKUBE_LDFLAGS="-X k8s.io/minikube/pkg/version.version=v1.15.1 -X k8s.io/minikube/pkg/version.isoVersion=v1.15.2-snapshot2 -X k8s.io/minikube/pkg/version.isoPath=minikube/iso -X k8s.io/minikube/pkg/version.gitCommitID="322e138f6bb1c47d7768efe58d62b18e4f4e54d4-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v3" ./test.sh 
= make lint =============================================================
ok
= go mod ================================================================
ok
= boilerplate ===========================================================
ok
= schema_check ==========================================================
ok
= go test ===============================================================
--- FAIL: TestDeleteAllProfiles (0.13s)
    delete_test.go:203: ListProfiles length = 9, expected 8
        valid: []
        invalid: [0xc000e3aab0 0xc000e3aae0 0xc000e3ab10 0xc000e3ab40 0xc000e3ab70 0xc000e3aba0 0xc000e3abd0 0xc000e3ac00 0xc000e3ac60]
E1128 10:50:03.679291   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=127.0.0.1:3128 to docker env.
E1128 10:50:03.679984   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=127.0.0.1:3128 to docker env.
E1128 10:50:03.680098   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=localhost:3128 to docker env.
E1128 10:50:03.680275   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=localhost:3128 to docker env.
E1128 10:50:03.680388   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://localhost:3128 to docker env.
E1128 10:50:03.680593   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://localhost:3128 to docker env.
E1128 10:50:03.680691   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://127.0.0.1:3128 to docker env.
E1128 10:50:03.680867   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://127.0.0.1:3128 to docker env.
FAIL
coverage: 18.7% of statements
FAIL	k8s.io/minikube/cmd/minikube/cmd	4.819s
ok  	k8s.io/minikube/cmd/minikube/cmd/config	0.084s	coverage: 21.0% of statements
ok  	k8s.io/minikube/pkg/addons	0.113s	coverage: 50.9% of statements
ok  	k8s.io/minikube/pkg/drivers	0.016s	coverage: 19.6% of statements
ok  	k8s.io/minikube/pkg/drivers/hyperkit	0.010s	coverage: 77.3% of statements
ok  	k8s.io/minikube/pkg/drivers/kic/oci	0.018s	coverage: 10.3% of statements
ok  	k8s.io/minikube/pkg/drivers/kvm	0.017s	coverage: 2.1% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper	0.726s	coverage: 55.1% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/bsutil	0.137s	coverage: 62.4% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/bsutil/ktmpl	0.012s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/images	0.004s	coverage: 82.0% of statements
ok  	k8s.io/minikube/pkg/minikube/cluster	0.082s	coverage: 14.5% of statements
ok  	k8s.io/minikube/pkg/minikube/command	0.061s	coverage: 4.2% of statements
ok  	k8s.io/minikube/pkg/minikube/config	0.104s	coverage: 72.7% of statements
ok  	k8s.io/minikube/pkg/minikube/cruntime	0.078s	coverage: 36.5% of statements
ok  	k8s.io/minikube/pkg/minikube/docker	0.054s	coverage: 20.8% of statements
ok  	k8s.io/minikube/pkg/minikube/driver	0.050s	coverage: 38.3% of statements
ok  	k8s.io/minikube/pkg/minikube/extract	0.035s	coverage: 58.6% of statements
ok  	k8s.io/minikube/pkg/minikube/image	0.048s	coverage: 3.0% of statements
ok  	k8s.io/minikube/pkg/minikube/kubeconfig	0.075s	coverage: 81.3% of statements
ok  	k8s.io/minikube/pkg/minikube/localpath	0.024s	coverage: 48.6% of statements
ok  	k8s.io/minikube/pkg/minikube/logs	0.105s	coverage: 1.3% of statements
ok  	k8s.io/minikube/pkg/minikube/machine	1.135s	coverage: 35.8% of statements
ok  	k8s.io/minikube/pkg/minikube/mustload	0.066s	coverage: 10.5% of statements
ok  	k8s.io/minikube/pkg/minikube/notify	0.054s	coverage: 92.9% of statements
ok  	k8s.io/minikube/pkg/minikube/out	0.021s	coverage: 66.7% of statements
ok  	k8s.io/minikube/pkg/minikube/out/register	0.026s	coverage: 55.4% of statements
ok  	k8s.io/minikube/pkg/minikube/perf	4.051s	coverage: 21.1% of statements
ok  	k8s.io/minikube/pkg/minikube/proxy	0.015s	coverage: 68.7% of statements
ok  	k8s.io/minikube/pkg/minikube/reason	0.034s	coverage: 80.0% of statements
ok  	k8s.io/minikube/pkg/minikube/registry	0.026s	coverage: 76.7% of statements
ok  	k8s.io/minikube/pkg/minikube/service	0.105s	coverage: 84.2% of statements
ok  	k8s.io/minikube/pkg/minikube/shell	0.007s	coverage: 94.3% of statements
ok  	k8s.io/minikube/pkg/minikube/storageclass	0.025s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/style	0.011s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/sysinit	0.057s	coverage: 6.0% of statements
ok  	k8s.io/minikube/pkg/minikube/translate	0.021s	coverage: 11.4% of statements
ok  	k8s.io/minikube/pkg/minikube/tunnel	1.766s	coverage: 64.2% of statements
ok  	k8s.io/minikube/pkg/util	0.610s	coverage: 75.7% of statements
ok  	k8s.io/minikube/pkg/util/lock	0.021s	coverage: 22.2% of statements
ok  	k8s.io/minikube/pkg/util/retry	0.003s	coverage: 0.0% of statements
FAIL
make: *** [test] Error 32

**Full output of `minikube start` command used, if not already included:**
[opc@kubernetes1 minikube]$ out/minikube start
😄  minikube v1.15.1 on Oracle 7.9
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=3800MB) ...
🐳  Preparing Kubernetes v1.19.4 on Docker 19.03.13 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
💡  kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

**My findings**

<details>

TestDeleteAllProfiles expects fixed number of 8 profiles which are taken from test data directory: "../../../pkg/minikube/config/testdata/delete-all".
However, extra profile "minikube" is added in ListProfiles method:
file:  pkg/minikube/config/profile.go

fragment responsible for adding:
```
// try to get profiles list based on all containers created by docker driver
	cs, err := oci.ListOwnedContainers(oci.Docker)
```
</details>

### PR DESCRIPTION
**Pull request by vishjain**
fixes #9795

### CODE REVIEW

**Comment by medyagh** [comment]
@vishjain  are u still working on this ?

**Comment by vishjain** [comment]
> @vishjain are u still working on this ?

Yes, please give me a few days to fix this. Out of town right now.

**Comment by vishjain** [comment]
@medyagh Please take a look when you get a chance. Thanks.

File: cmd/minikube/cmd/delete.go [hunk 1]
```diff
@@ -87,6 +87,24 @@ func (error DeletionError) Error() string {
 	return error.Err.Error()
 }
 
+var DeleteHostAndDirectoriesGetter = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
```
Comment (medyagh) [inline cmd/minikube/cmd/delete.go:90]: this func is not used anywhere in the code, lets make it private, also name Getter doesnt sound right here.

how about
var hostAndDirsDeleter = func ( ...

**Review by medyagh — CHANGES_REQUESTED** [review]

**Review by medyagh — CHANGES_REQUESTED** [review]
please see comment

**Comment by vishjain** [comment]
@medyagh Made changes. Thanks.

File: cmd/minikube/cmd/delete_test.go [hunk 2]
```diff
@@ -154,6 +158,17 @@ func TestDeleteProfile(t *testing.T) {
 	}
 }
 
+var DeleteHostAndDirectoriesMock = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
```
Comment (medyagh) [inline cmd/minikube/cmd/delete_test.go:161]: this needs to be private as well and rename to the same name as the other with "hostAndDirsDeleterMock"

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by vishjain** [comment]
@medyagh All names should be fixed. Thanks!

**Comment by vishjain** [comment]
@medyagh Can you please approve this/review this?

**Review by medyagh — APPROVED** [review]
thank you @vishjain

**Pull request merged**